tests/plugins/test_default_qubit_autograd.py

```diff
@@ -0,0 +1,341 @@
+
+    @pytest.mark.parametrize("operation", [qml.U3, qml.U3.decomposition])
+    @pytest.mark.parametrize("diff_method", ["backprop", "parameter-shift", "finite-diff"])
+    def test_autograd_interface_gradient(self, operation, diff_method, tol):
+        """Tests that the gradient of an arbitrary U3 gate is correct
+        using the Autograd interface, using a variety of differentiation methods."""
+        dev = qml.device("default.qubit.autograd", wires=1)
+
+        @qml.qnode(dev, diff_method=diff_method, interface="autograd")
+        def circuit(x, weights, w=None):
+            """In this example, a mixture of scalar
+            arguments, array arguments, and keyword arguments are used."""
+            qml.QubitStateVector(1j * np.array([1, -1]) / np.sqrt(2), wires=w)
+            operation(x, weights[0], weights[1], wires=w)
+            return qml.expval(qml.PauliX(w))
+
+        # Check that the correct QNode type is being used.
+        if diff_method == "backprop":
+            assert isinstance(circuit, qml.qnodes.PassthruQNode)
+            assert not hasattr(circuit, "jacobian")
+        else:
+            assert not isinstance(circuit, qml.qnodes.PassthruQNode)
+            assert hasattr(circuit, "jacobian")
+
+        def cost(params):
+            """Perform some classical processing"""
+            return circuit(params[0], params[1:], w=0) ** 2
+
+        theta = 0.543
+        phi = -0.234
+        lam = 0.654
+
+        params = np.array([theta, phi, lam], requires_grad=True)
+
+        res = cost(params)
+        expected_cost = (np.sin(lam) * np.sin(phi) - np.cos(theta) * np.cos(lam) * np.cos(phi)) ** 2
+        assert np.allclose(res, expected_cost, atol=tol, rtol=0)
+
+        res = qml.grad(cost)(params)
+        expected_grad = (
+            np.array(
+                [
+                    np.sin(theta) * np.cos(lam) * np.cos(phi),
+                    np.cos(theta) * np.cos(lam) * np.sin(phi) + np.sin(lam) * np.cos(phi),
+                    np.cos(theta) * np.sin(lam) * np.cos(phi) + np.cos(lam) * np.sin(phi),
+                ]
+            )
+            * 2
+            * (np.sin(lam) * np.sin(phi) - np.cos(theta) * np.cos(lam) * np.cos(phi))
+        )
+        assert np.allclose(res, expected_grad, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("interface", ["tf", "torch"])
+    def test_error_backprop_wrong_interface(self, interface, tol):
+        """Tests that an error is raised if diff_method='backprop' but not using
+        the Autograd interface"""
+        dev = qml.device("default.qubit.autograd", wires=1)
+
+        def circuit(x, w=None):
+            qml.RZ(x, wires=w)
+            return qml.expval(qml.PauliX(w))
+
+        with pytest.raises(
+            ValueError,
+            match="default.qubit.autograd only supports diff_method='backprop' when using the autograd interface",
+        ):
+            qml.qnode(dev, diff_method="backprop", interface=interface)(circuit)

```

Comment:
It looks like it would be really easy to parametrize this and similar tests over a few more numerical test cases

Reply:
I considered that. I wasn't sure if it was worth it in this particular case, as the logic being tested is independent of operation parameter values. So it would increase test run time without picking up edge cases